antefoyer/tests/test_antefoyer.py

```diff
@@ -17,65 +18,88 @@
+
 
 @pytest.mark.skipif(ANTECHAMBER is None, reason="antechamber is not installed")
 def test_invalid_structure():
-    ethane = pmd.load_file(get_fn('ethane.mol2'),structure=True)
-    with pytest.raises(FoyerError,match=r"Unknown molecule format.*"):
-        typed = ante_atomtyping(ethane.topology,'gaff')
+    ethane = pmd.load_file(get_fn("ethane.mol2"), structure=True)
+    with pytest.raises(FoyerError, match=r"Unknown molecule format.*"):
+        typed = ante_atomtyping(ethane.topology, "gaff")
+
 
 @pytest.mark.skipif(ANTECHAMBER is None, reason="antechamber is not installed")
 @pytest.mark.skipif(not has_mbuild, reason="mbuild is not installed")
 def test_convert_mbuild():
     import mbuild as mb
-    ethane = mb.load(get_fn('ethane.mol2'))
-    typed = ante_atomtyping(ethane,'gaff')
+
+    ethane = mb.load(get_fn("ethane.mol2"))
+    typed = ante_atomtyping(ethane, "gaff")
+
 
 @pytest.mark.skipif(ANTECHAMBER is None, reason="antechamber is not installed")
 def test_multiple_molecules():
-    ethane = pmd.load_file(get_fn('ethane.mol2'),structure=True)
+    ethane = pmd.load_file(get_fn("ethane.mol2"), structure=True)
     # Remove a bond to break the molecule apart
     ethane.bonds.remove(ethane.bonds[0])
-    with pytest.raises(FoyerError,match=r"Antechamber requires connectivity.*"):
-        typed = ante_atomtyping(ethane,'gaff')
+    with pytest.raises(FoyerError, match=r"Antechamber requires connectivity.*"):
+        typed = ante_atomtyping(ethane, "gaff")
 
-# TODO: Write this test. What to do about error logfile handling? 
+
+# TODO: Write this test. What to do about error logfile handling?
 @pytest.mark.skipif(ANTECHAMBER is None, reason="antechamber is not installed")
 def test_ante_error():
-    ethane = pmd.load_file(get_fn('ethane.mol2'),structure=True)
+    ethane = pmd.load_file(get_fn("ethane.mol2"), structure=True)
     with temporary_directory() as tmpdir:
         with temporary_cd(tmpdir):
-            with pytest.raises(RuntimeError,match=r"Antechamber failed"):
-                charges = ante_charges(ethane,'bcc',net_charge=-1)
-            assert isfile('ante_errorlog.txt')
+            with pytest.raises(RuntimeError, match=r"Antechamber failed"):
+                charges = ante_charges(ethane, "bcc", net_charge=-1)
+            assert isfile("ante_errorlog.txt")
+
+
+@pytest.mark.skipif(ANTECHAMBER is None, reason="antechamber is not installed")
+def test_ante_charge_delta():
+    ethane = pmd.load_file(get_fn("ethane.mol2"), structure=True)
+    charges = ante_charges(ethane, "bcc", net_charge=0)
 
+    assert np.allclose(sum([i.charge for i in charges]), 0, atol=5e-3)
+
+
+@pytest.mark.skipif(ANTECHAMBER is None, reason="antechamber is not installed")
+def test_charge_tolerance():
+    with pytest.raises(ValueError, match=r"The sum of charges"):
+        ethane = pmd.load_file(get_fn("ethane.mol2"), structure=True)
+        ante_charges(ethane, "bcc", charge_tol=0.001)
```

Comment:
Curiosity? How far off are the net charges on Ethane?

Reply:
Ethane is off by `0.0020000000000000087`.